### ISSUE TITLE
Error parser raises unexpected error when error line is empty

### ISSUE THREAD

**Issue by user202729** [charlesstaats/asypictureB#2]
Example document:
```tex
\documentclass{article}
\usepackage{asypictureB}
\begin{document}
\begin{asypicture}{name=foo}
  label("a);
\end{asypicture}
\end{document}
```

Content of `a-foo_errors.txt`:
```

^
a-foo.asy: 6.1: string not terminated

error: could not load module 'a-foo.asy'
```

Resulting TeX error:
```
Runaway argument?
! Paragraph ended before \@asy@processerrorline was complete.
<to be read again> 
                   \par 
l.6 \end{asypicture}                
```
Expected behavior: that `asypictureB` cleanly wraps the Asymptote error.

**Comment by charlesstaats** [comment]
I've changed one particular instance of the line `\endlinechar=‘ˆˆJ` to `\endlinechar=-1`. This has eliminated several gratuitous empty lines in the error message, and appears to make the `\long` declaration unnecessary. Let me know if it works for you.

I looked into `\read` versus `\readline`, and there's no clear consensus that `\readline` is better. To avoid the programming issues you mentioned, I've continued to use `\read`. If there are other reasons to prefer `\readline`, I'm happy to hear them. (If the implications are something like a minor misformatting of the error message, I'm still interested to know about it, but I reserve the right to deprioritize fixing it.)
